phpBB/phpbb/install/controller/helper.php

```diff
@@ -0,0 +1,364 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\install\controller;
+
+use phpbb\install\helper\navigation\navigation_provider;
+use phpbb\language\language;
+use phpbb\language\language_file_helper;
+use phpbb\path_helper;
+use phpbb\request\request;
+use phpbb\request\request_interface;
+use phpbb\routing\router;
+use phpbb\symfony_request;
+use phpbb\template\template;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Cookie;
+
+/**
+ * A duplicate of \phpbb\controller\helper
+ *
+ * This class is necessary because of controller\helper's legacy function calls
+ * to page_header() page_footer() functions which has unavailable dependencies.
+ */
+class helper
+{
+	/**
+	 * @var \phpbb\language\language
+	 */
+	protected $language;
+
+	/**
+	 * @var bool|string
+	 */
+	protected $language_cookie;
+
+	/**
+	 * @var \phpbb\language\language_file_helper
+	 */
+	protected $lang_helper;
+
+	/**
+	 * @var \phpbb\install\helper\navigation\navigation_provider
+	 */
+	protected $navigation_provider;
+
+	/**
+	 * @var \phpbb\template\template
+	 */
+	protected $template;
+
+	/**
+	 * @var \phpbb\path_helper
+	 */
+	protected $path_helper;
+
+	/**
+	 * @var \phpbb\request\request
+	 */
+	protected $phpbb_request;
+
+	/**
+	 * @var \phpbb\symfony_request
+	 */
+	protected $request;
+
+	/**
+	 * @var \phpbb\routing\router
+	 */
+	protected $router;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_admin_path;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * Constructor
+	 *
+	 * @param language				$language
+	 * @param language_file_helper	$lang_helper
+	 * @param navigation_provider	$nav
+	 * @param template				$template
+	 * @param path_helper			$path_helper
+	 * @param request				$phpbb_request
+	 * @param symfony_request		$request
+	 * @param router				$router
+	 * @param string				$phpbb_root_path
+	 */
+	public function __construct(language $language, language_file_helper $lang_helper, navigation_provider $nav, template $template, path_helper $path_helper, request $phpbb_request, symfony_request $request, router $router, $phpbb_root_path)
+	{
+		$this->language = $language;
+		$this->language_cookie = false;
+		$this->lang_helper = $lang_helper;
+		$this->navigation_provider = $nav;
+		$this->template = $template;
+		$this->path_helper = $path_helper;
+		$this->phpbb_request = $phpbb_request;
+		$this->request = $request;
+		$this->router = $router;
+		$this->phpbb_root_path = $phpbb_root_path;
+		$this->phpbb_admin_path = $phpbb_root_path . 'adm/';
+
+		$this->handle_language_select();
```

Comment:
that's because for now the request lives in the container, but one time it will change and we will pass a request_stack object instead and the request will be set by the http_kernel. Having the request in the container can be easy at first but it's a non sense and it also make things a lot more difficult sometimes (it has been discussed a lot in Symfony)